compiler/rustc_trait_selection/src/opaque_types.rs

```diff
@@ -38,13 +38,16 @@ pub struct OpaqueTypeDecl<'tcx> {
     /// then `substs` would be `['a, T]`.
     pub substs: SubstsRef<'tcx>,
 
-    /// The span of this particular definition of the opaque type.  So
+    /// The span of this particular definition of the opaque type. So
     /// for example:
     ///
     /// ```
     /// type Foo = impl Baz;
```

Comment:
You'll also need `trait Baz {}` I think.

Reply:
Weirdly, CI passed...

But I'll add it anyway.

Reply:
I don't think CI runs doc-tests on the compiler itself.

Reply:
Hmm, maybe it should!

Reply:
I am personally not sure on if we want to run doc tests on rustc itself. I feel like rustc can live with examples being incomplete,
so not defining `Baz` here feels like a good thing to me as I don't think it would make it easier to understand this example.